test/k8sT/Chaos.go

```diff
@@ -59,103 +53,211 @@ var _ = Describe("K8sChaosTest", func() {
-				res := kubectl.ExecPodCmd(
-					helpers.DefaultNamespace, pod, helpers.Ping(ip))
-				log.Debugf("Pod %s ping %v", pod, ip)
-				ExpectWithOffset(1, res).To(helpers.CMDSuccess(),
-					"Cannot ping from %q to %q", pod, ip)
-
-				By("Curling testds-service via ClusterIP %q", host)
-				res = kubectl.ExecPodCmd(
-					helpers.DefaultNamespace, pod, helpers.CurlFail("http://%s:80/", host))
-				ExpectWithOffset(1, res).To(helpers.CMDSuccess(),
-					"Cannot curl from %q to testds-service via ClusterIP", pod)
-
-				By("Curling testds-service via DNS hostname")
-				res = kubectl.ExecPodCmd(
-					helpers.DefaultNamespace, pod, helpers.CurlFail("http://%s:80/", testDSService))
-				ExpectWithOffset(1, res).To(helpers.CMDSuccess(),
-					"Cannot curl from %q to testds-service via DNS hostname", pod)
+	Context("Connectivity demo application", func() {
+		BeforeEach(func() {
+			kubectl.Apply(demoDSPath).ExpectSuccess("DS deployment cannot be applied")
+
+			err := kubectl.WaitforPods(
+				helpers.DefaultNamespace, fmt.Sprintf("-l zgroup=testDS"), 300)
+			Expect(err).Should(BeNil(), "Pods are not ready after timeout")
+		})
+
+		AfterEach(func() {
+			kubectl.Delete(demoDSPath).ExpectSuccess(
+				"%s deployment cannot be deleted", demoDSPath)
+			ExpectAllPodsTerminated(kubectl)
+
+		})
+
+		PingService := func() {
+			pods, err := kubectl.GetPodNames(helpers.DefaultNamespace, "zgroup=testDSClient")
+			Expect(err).To(BeNil(), "Cannot get pods names")
+			Expect(len(pods)).To(BeNumerically(">", 0), "No pods available to test connectivity")
+
+			dsPods, err := kubectl.GetPodsIPs(helpers.DefaultNamespace, "zgroup=testDS")
+			Expect(err).To(BeNil(), "Cannot get daemonset pods IPS")
+			Expect(len(pods)).To(BeNumerically(">", 0), "No pods available to test connectivity")
+
+			By("Waiting for kube-dns entry for service testds-service")
+			err = kubectl.WaitForKubeDNSEntry(testDSService, helpers.DefaultNamespace)
+			ExpectWithOffset(1, err).To(BeNil(), "DNS entry is not ready after timeout")
+
+			By("Getting ClusterIP For testds-service")
+			host, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, "testds-service")
+			ExpectWithOffset(1, err).To(BeNil(), "unable to get ClusterIP and port for service testds-service")
+
+			for _, pod := range pods {
+				for _, ip := range dsPods {
+					By("Pinging test-ds service pod with IP %q from client pod %q", ip, pod)
+					res := kubectl.ExecPodCmd(
+						helpers.DefaultNamespace, pod, helpers.Ping(ip))
+					log.Debugf("Pod %s ping %v", pod, ip)
+					ExpectWithOffset(1, res).To(helpers.CMDSuccess(),
+						"Cannot ping from %q to %q", pod, ip)
+
+					By("Curling testds-service via ClusterIP %q", host)
+					res = kubectl.ExecPodCmd(
+						helpers.DefaultNamespace, pod, helpers.CurlFail("http://%s:80/", host))
+					ExpectWithOffset(1, res).To(helpers.CMDSuccess(),
+						"Cannot curl from %q to testds-service via ClusterIP", pod)
+
+					By("Curling testds-service via DNS hostname")
```

Comment:
Not-blocking: this could be done outside of the inner for loop because you aren't connecting to the `ip` explicitly here. Same for the "Curling testds-service via ClusterIP" above.